src/main/java/emu/grasscutter/data/excels/GadgetInteractData.java

```diff
@@ -0,0 +1,42 @@
+package emu.grasscutter.data.excels;
+
+import emu.grasscutter.data.GameResource;
+import emu.grasscutter.data.ResourceType;
+import lombok.Data;
+import lombok.Getter;
+
+@ResourceType(name = "GadgetInteractExcelConfigData.json")
+@Getter
+public class GadgetInteractData extends GameResource {
+    private int interactId;
+    private String actionType;
+    private int param1;
+    private GadgetInteractActionList[] actionList;
+    private GadgetInteractCostItems[] costItems;
+    private long uiTitleTextMapHash;
+    private long uiDescTextMapHash;
+    private GadgetInteractCondList[] condList;
+    private Boolean consumeItemNum;
+
+    @Data
+    public static class GadgetInteractActionList {
+        private String actionType;
+        private int[] param;
+    }
+
+    @Data
+    public static class GadgetInteractCostItems {
```

Comment:
since this is one entry of many, it might be better to call it `GadgetInteractCostItem`